### PR TITLE
Add SSL option to nri-jmx

### PR DESCRIPTION
**Pull request by vvydier**
Adding SSL option to nri-jmx invovles changes to infra-integrations-sdk and nrjmx (java) projects
Here are the related PR's -> 
https://github.com/newrelic/infra-integrations-sdk/pull/163
https://github.com/newrelic/nrjmx/pull/11

I have create a new function jmx.OpenWIthSSL so that it does not impact older non-SSL JMX integrations

### CODE REVIEW

**Comment by varas** [comment]
SDK PR moved to https://github.com/newrelic/infra-integrations-sdk/pull/165

File: jmx-ssl-config.yml.sample [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+instances:
+  - name: jmx
+    command: all_data
+    arguments:
+      jmx_host: jmx-host.localnet
+      jmx_port: 9999
+      jmx_user: admin
+      jmx_pass: admin
+      key_store: /etc/pki/JMXClientKeyStore.key
```
Comment (cpheps) [inline jmx-ssl-config.yml.sample]: Is it possible to use the default java truststore or does nrjmx need to be fed a truststore?

Comment (vvydier) [reply jmx-ssl-config.yml.sample]: yes, you can.. its optional. If you use the default truststore, any user can make the JMX query over SSL.

Comment (cpheps) [reply jmx-ssl-config.yml.sample]: Do we need to use the `OpenWithSSL` from the sdk to use that? Looks like you need all the key store and trust store variables set in order to use the ssl path.

Comment (vvydier) [reply jmx-ssl-config.yml.sample]: I could have changed Jmx.Open to use variadic arguments for Jmx.Open but I did not want to impact any older JMX integrations that use Jmx.Open
nrjmx (java) does not use the SSL arguments if its not supplied. nri-jmx wrapper uses Jmx.Open which has 4 arguments that I have left unchanged, so I don't impact other JMX integrations

File: vendor/vendor.json [hunk 2]
```diff
@@ -32,54 +32,6 @@
 			"revision": "e2ffdb16a802fe2bb95e2e35ff34f0e53aeef34f",
 			"revisionTime": "2018-05-06T08:24:08Z"
 		},
-		{
```
Comment (ccheek21) [inline vendor/vendor.json]: It appears you have removed the SDK from vendoring. We try to keep all dependencies up to date in the vendor folder.

**Comment by vvydier** [comment]
> Could we update the change log with the version bump and notes?

I updates version number in CHANGELOG.md and src/jmx.go to 1.0.2

File: vendor/vendor.json (hunk 2 above)
Comment (vvydier) [reply vendor/vendor.json]: I have added it with  `govendor update +external github.com/newrelic/infra-integrations-sdk/...` and tested it locally

**Pull request merged**